File: 52936349-main/pset6/readability.py
from cs50 import get_string
import re

txt = get_string("Write your text: ")

#letters
letters = sum(c.isalpha() for c in txt) #sum of all letters in text
#print(f"Amount of letters in text: {letters}")

#words
words = len(txt.split())
#print(f"Amount of words in text: {words}")

#sentences
sentences = txt.count('.') + txt.count('!') + txt.count('?')
#print(f"Amount of sentences in text: {sentences}")

#Coleman-Liau index
l = letters / words * 100
s = sentences / words * 100

index = (0.0588 * l - 0.296 * s - 15.8)
index = round(index)
#if index is 16 or higher, print "Grade 16+"
if index > 16:
    print("Grade 16+")
# if index is less than 1, print "Before Grade 1"
elif index > 1 and index < 17:
    print(f"Grade {index}")
#if the index is less than 1, print "Before Grade 1"
else:
    print("Before Grade 1")

File: 52936349-main/pset2/Readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    int letters = 0;
    int sentences = 0;
    int words = 1;

    string name = get_string("Write your text here: ");

    for(int i = 0; i < strlen(name); i++)
    {
        if (isalpha(name[i])) // if the character in the array in a letter - +1 letter counter
        {
            letters += 1;
        }
        if (name[i] == '.' || name[i] == '?' || name[i] == '!')
        {
            sentences += 1;
        }
        if (name[i] == (' ')) // everytime there is a space between words, counter +1 for words
        {
            words += 1;
        }
    }
    // printf("%i\n", letters);
    // printf("%i\n", sentences);
    // printf("%i\n", words);

    //  Coleman-Liau index

    float L = ((float)letters / (float)words * 100);
    float S = ((float)sentences / (float)words * 100);

    float index = (0.0588 * L - 0.296 * S - 15.8);
    printf("Grade %.f\n", round(index));


    if (index < 1)
    {
        printf("Before grade 1\n");
    }
}
